From f48bdf59117e28231a2f1eb7dd65ecbab0c2f5ca Mon Sep 17 00:00:00 2001
From: Rex Lu <rex.lu@mediatek.com>
Date: Thu, 2 Oct 2025 09:29:54 +0800
Subject: [PATCH 071/105] mtk: mt76: mt7996: Refactor L1 SER flow

remove unnecessary actions during L1, some actions will cause L1 SER processing time too long.
ex : mt76_abort_scan and mt76_abort_roc will increase processing time from 0.16s to 7s.

Signed-off-by: Rex Lu <rex.lu@mediatek.com>
---
 mt7996/mac.c | 14 +++-----------
 1 file changed, 3 insertions(+), 11 deletions(-)

diff --git a/mt7996/mac.c b/mt7996/mac.c
index 7956ee652..e6dfce313 100644
--- a/mt7996/mac.c
+++ b/mt7996/mac.c
@@ -2494,15 +2494,10 @@ void mt7996_mac_reset_work(struct work_struct *work)
 
 	set_bit(MT76_RESET, &dev->mphy.state);
 	set_bit(MT76_MCU_RESET, &dev->mphy.state);
-	mt76_abort_scan(&dev->mt76);
 	wake_up(&dev->mt76.mcu.wait);
 
-	cancel_work_sync(&dev->wed_rro.work);
-	mt7996_for_each_phy(dev, phy) {
-		mt76_abort_roc(phy->mt76);
+	mt7996_for_each_phy(dev, phy)
 		set_bit(MT76_RESET, &phy->mt76->state);
-		cancel_delayed_work_sync(&phy->mt76->mac_work);
-	}
 
 	mt76_worker_disable(&dev->mt76.tx_worker);
 
@@ -2611,11 +2606,8 @@ void mt7996_mac_reset_work(struct work_struct *work)
 
 	mt7996_npu_hw_init(dev);
 
-	mt7996_for_each_phy(dev, phy)
-		ieee80211_queue_delayed_work(hw, &phy->mt76->mac_work,
-					     MT7996_WATCHDOG_TIME);
-
-	ieee80211_queue_delayed_work(mt76_hw(dev), &dev->scs_work, HZ);
+	dev_info(dev->mt76.dev,"\n%s L1 SER recovery completed.",
+		 wiphy_name(dev->mt76.hw->wiphy));
 }
 
 /* firmware coredump */
-- 
2.45.2

